File: fortran/src/variables.f90
module variables_mod
    implicit none

    ! Module-level constants
    integer, parameter :: MAX_SCORE = 100
    real, parameter :: PI = 3.14159

contains
    subroutine demonstrate_variables()
        ! Variable declarations
        character(len=20) :: name
        integer :: age
        real :: height
        logical :: is_student
        
        ! Arrays
        integer, dimension(5) :: numbers = [1, 2, 3, 4, 5]
        real, dimension(2,2) :: matrix
        
        ! Derived type (similar to struct)
        type :: person
            character(len=20) :: name
            integer :: age
        end type person
        type(person) :: student
        
        ! Assigning values
        name = 'John Doe'
        age = 25
        height = 1.75
        is_student = .true.
        matrix = reshape([1.0, 2.0, 3.0, 4.0], [2,2])
        
        ! Output
        print *, '=== Variables and Data Types ==='
        print *, 'Name: ', trim(name)
        print *, 'Age: ', age
        print *, 'Height: ', height
        print *, 'Is Student: ', is_student
        
        print *, 'Numbers array: ', numbers
        print *, 'Matrix:'
        print *, matrix(1,:)
        print *, matrix(2,:)
        
        ! Using derived type
        student%name = 'Jane Smith'
        student%age = 20
        print *, 'Student:', trim(student%name), ', Age:', student%age
        
        ! Using constant
        print *, 'Maximum Score:', MAX_SCORE
        print *, 'PI:', PI
    end subroutine demonstrate_variables
end module variables_mod

File: ruby/lib/variables.rb
module Variables
  # Constants should be defined at module level
  MAXIMUM_SCORE = 100
  
  # Class variables should also be at module level
  @@class_variable = "Class"
  
  def self.demonstrate
    puts "\n=== Variables and Data Types ==="
    
    # Different types of variables
    name = "John Doe"       # String
    age = 25               # Integer
    height = 1.75         # Float
    is_student = true     # Boolean
    
    # Symbol - a common Ruby type
    status = :active
    
    # Arrays and Hashes
    hobbies = ["reading", "coding"]
    user_info = { name: "John", age: 25 }
    
    # Displaying variables with their types
    puts "Name: #{name} (#{name.class})"
    puts "Age: #{age} (#{age.class})"
    puts "Height: #{height} (#{height.class})"
    puts "Is Student: #{is_student} (#{is_student.class})"
    puts "Status: #{status} (#{status.class})"
    puts "Hobbies: #{hobbies} (#{hobbies.class})"
    puts "User Info: #{user_info} (#{user_info.class})"
    
    # Displaying constant
    puts "\nConstants:"
    puts "Maximum Score: #{MAXIMUM_SCORE}"
    
    # Variable scope indicators
    @instance_variable = "Instance"
    $global_variable = "Global"
    local_variable = "Local"
    
    puts "\nVariable Scopes:"
    puts "Instance: #{@instance_variable}"
    puts "Class: #{@@class_variable}"
    puts "Global: #{$global_variable}"
    puts "Local: #{local_variable}"
  end
end